src/components/FormField/SelectContext.tsx

```diff
@@ -0,0 +1,101 @@
+import {
+  createContext,
+  HTMLAttributes,
+  ReactElement,
+  ReactNode,
+  useContext,
+  useEffect,
+  useState,
+} from "react";
+
+const SelectValue = (props: HTMLAttributes<HTMLDivElement>) => <div {...props} />;
+
+type ContextProps = {
+  selectedValue?: string | null;
+  onSelect: (value: string) => void;
+  popperOpen: boolean;
+  valueNode: ReactNode | null;
+  onOpenChange: (value: boolean) => void;
+};
+
+const SelectContext = createContext<ContextProps>({
+  selectedValue: undefined,
+  onSelect: () => null,
+  popperOpen: false,
+  valueNode: null,
+  onOpenChange: () => null,
+});
+
+type Props = {
+  children: ReactNode;
+  value?: string;
+  defaultValue?: string;
+  updateValueNode: (value?: string) => ReactElement | null;
+  defaultOpen?: boolean;
+  onOpenChange?: (value: boolean) => void;
+  onChange?: (value: string) => void;
+};
+
+export const SelectProvider = ({
+  value,
+  children,
+}: {
+  children: ReactNode;
+  value: ContextProps;
+}) => {
+  return <SelectContext.Provider value={value}>{children}</SelectContext.Provider>;
+};
+
+export const SelectContextProvider = ({
```

Comment:
why do we need a provider for a simple select?

Reply:
This is to process the onSelect values and also setting the value in the trigger element and to toggle the open state when clicking elements
I have gone through the radix-ui select code too then are also using context for toggling the state and using DocumentFragment and createPortal to set value in the select trigger
I thought that was expensive and used this method